src/Aspire.Hosting/ApplicationModel/IContainerRegistry.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents container registry information for deployment targets.
+/// </summary>
+public interface IContainerRegistry
+{
+    /// <summary>
+    /// Gets the name of the container registry.
+    /// </summary>
+    IManifestExpressionProvider Name { get; }
+
+    /// <summary>
+    /// Gets the endpoint URL of the container registry.
+    /// </summary>
+    IManifestExpressionProvider Endpoint { get; }
+
+    /// <summary>
+    /// Gets the managed identity ID associated with the container registry.
+    /// </summary>
+    IManifestExpressionProvider ManagedIdentityId { get; }
```

Comment:
This is an azure concept, so I think layering wise, we need to rename this `AuthenticationInfo` or something more generic. Or we need to have multiple interfaces:

```C#
public interface IContainerRegistry
{
    IManifestExpressionProvider Endpoint { get; }
}

public interface IAzureContainerRegistry : IContainerRegistry
{
    BicepOutputReference Name { get; }
    BicepOutputReference Endpoint { get; }
    BicepOutputReference ManagedIdentityId { get; }
    
    IContainerRegistry.Endpoint => Endpoint;
}
```

Reply:
> Or we need to have multiple interfaces

I think having multiple interfaces (i.e. an IAzureContainerRegistry) would help here. I'm not sure we could model an effective `AuthenticationInfo`, or something generic.

Reply:
> This is an azure concept, so I think layering wise, we need to rename this AuthenticationInfo or something more generic. Or we need to have multiple interfaces:

Does this mean that `IContainerRegistry` is defined in `Aspire.Hosting` and `IAzureContainerRegistry` is define in `Aspire.Hosting.Azure.AppContainers`?